Day-II-Calculator/calculator/src/components/ButtonComponents/CalcButton.js

```diff
@@ -0,0 +1,20 @@
+import React from 'react';
+
+function handler(label, onClick) {
+	return () => onClick(label);
+}
+
+function CalcButton(props) {
+  let className = 'calc-button';
+  if (props.large) {
+  	className += ' large';
+  }
+
+  return (
+	<React.Fragment>
+    	<button className={className} onClick={handler(props.label, props.onClick)} >{props.label}</button>
```

Comment:
I'd consider either using [object destructuring]() to shorten this a bit or breaking it up onto multiple lines (or both). For example:
```javascript
function CalcButton({ label, large, onClick }) {
  let className = 'calc-button';
  if (large) className += ' large';

  return (
    <button
      className={className}
      onClick={handler(label, onClick)}
    >
      {label}
    </button>
  );
}
```